leshan-server-cluster/src/main/java/org/eclipse/leshan/server/cluster/serialization/RegistrationSerDes.java

```diff
@@ -86,8 +85,7 @@ public static String sSerialize(Registration r) {
 
     public static Registration deserialize(JsonObject jObj) {
         Registration.Builder b = new Registration.Builder(jObj.getString("regId", null), jObj.getString("ep", null),
-                new InetSocketAddress(jObj.getString("address", null), jObj.getInt("port", 0)).getAddress(),
-                jObj.getInt("port", 0),
+                IdentitySerDes.deserialize(jObj.get("identity").asObject()),
```

Comment:
v.get("identity"): same NPE issue with already serialized observations?
We should handle the transition to build the identity from the "address" and "port" fields.